src/Microsoft.AspNetCore.ResponseCaching/Interfaces/IResponseCachingCacheKeyModifier.cs

```diff
@@ -5,13 +5,20 @@
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
-    public interface IResponseCachingCacheKeySuffixProvider
+    public interface IResponseCachingCacheKeyModifier
     {
         /// <summary>
+        /// Create a key segment that is prepended to the default cache key.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
+        /// <returns>The key segment that will be prepended to the default cache key.</returns>
+        string CreatKeyPrefix(HttpContext httpContext);
```

Comment:
A few interactions we might need to consider here:
1) Hashing
Not sure if this is something we want to implement yet but if we do, do we want to keep the prefix unhashed for ease of identification and the suffix to be hashed along with the rest of the key?
2) Base key vs VaryBy key
This won't affect the user but do we the suffix to be in the base key or a part of the vary-by key?

I agree, functionally there's no reason to have both a prefix and a suffix other than aesthetics when inspecting the cache, but I thought that was the point?

We can also add a default service for creating keys and let the user override that.